### PR TITLE
Fix section gradebook / grade by section error.

### LINKED ISSUE #112: Grading submissions by section gives 500 Error
Problem with SQL query at app/controllers/assessment/grading.rb:625-626.

### PR DESCRIPTION
**Pull request by ymzong**
Fixes #112.

This fixes both Grade by Section and Section Gradebook.

### CODE REVIEW

File: app/controllers/assessment/grading.rb [hunk 1]
```diff
@@ -622,8 +622,8 @@ def load_gradesheet_data
   
     # section filter
     o = params[:section] ? {
-      :conditions => { :assessment_id => id, :users => { :section => @user.section } },
-      :joins => "INNER JOIN users ON submissions.user_id = users.id" # hash doesn't work, Rails = stupid
+      :conditions => { :assessment_id => id, :course_user_data => { :section => @cud.section } },
+      :joins => "INNER JOIN course_user_data ON submissions.course_user_datum_id = course_user_data.id"
```
Comment (dlbucci) [inline app/controllers/assessment/grading.rb]: This line isn't even ever used, so you should remove it.  Other than that, if this is tested to work, you can pull it.

File: app/controllers/assessment/grading.rb [hunk 2]
```diff
@@ -622,8 +622,8 @@ def load_gradesheet_data
   
     # section filter
     o = params[:section] ? {
-      :conditions => { :assessment_id => id, :users => { :section => @user.section } },
-      :joins => "INNER JOIN users ON submissions.user_id = users.id" # hash doesn't work, Rails = stupid
+      :conditions => { :assessment_id => id, :course_user_data => { :section => @cud.section } },
```
Comment (dlbucci) [inline app/controllers/assessment/grading.rb]: I tried this and it complained about there being no course_user_data.section column.  Are you sure this works?

Comment (ymzong) [inline app/controllers/assessment/grading.rb]: @dlbucci It worked locally when I tested it -- I'll give it another look.

Comment (ymzong) [inline app/controllers/assessment/grading.rb]: @dlbucci How did you get "no `course_user_data.section` column" error? Are you trying to do `bulkUpload`?

**Pull request merged**